### PR TITLE
Feature: Installation Appendix Style Refresh

### LINKED ISSUE #3468: Feature: Installation Appendix Style Refresh
### Why?
The [installations appendix](https://www.theodinproject.com/lessons/installation_lessons) is just a list of un styled links, we can do better!

### Acceptance Criteria
We don't need anything too fancy, but turning each list item into a box with an icon and good hover effects will go a long way.

Just some ideas I've had for it:
* A layout [like this](https://tailwindui.com/components/application-ui/layout/list-containers#component-a0f475f9acf9194cd19255e19acde240).
* For icons, we could go with something generic from [heroicons](https://heroicons.com/)
* Another option for icons would be using the branding that represents the tech for each lesson. For example the node logo for the node installation lesson, ruby logo for ruby install etc.  SVG's for those can be sourced from [here](https://simpleicons.org/). If we go this route, we will need to add an `installation_icon` column to the lessons table which will reference the path to the icon in the assets. Check out the `badge_uri` attribute on courses and paths for an example of this.

Additional
To make this page easier to find and clearer
* Rename "Appendix" footer item to "Installation Guides"
* Change the title on the page from "Installation Lessons Appendix" to "Installation Guides"
* Does the installation overview lesson make sense within the context of this page? can we remove it?

### PR DESCRIPTION
**Pull request by wzhengg**
## Because

The appearance of the installation guides page will be more modern and up to date with the rest of the website, and the page will be easier to find.

## This PR

- Change list items from mostly unstyled links to cards with hover effects
- Add icon to each list item
- Rename page title and footer link to "Installation Guides"

## Issue

Closes #3468 

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `keyword: brief description of change` format, using one of the following keywords:
  - `Feature` - adds new or amends existing user-facing behavior
  - `Chore` - changes that have no user-facing value, refactors, dependency bumps, etc
  - `Fix` - bug fixes
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] I have verified all tests and linters pass after making these changes.
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [ ] If applicable, this PR includes new or updated automated tests

### CODE REVIEW

File: app/views/lessons/installation_lessons/index.html.erb [hunk 1]
```diff
@@ -1,14 +1,20 @@
...
+<div class="page-container min-h-screen">
+  <h1 class="page-heading-title text-center">Installation Guides</h1>
   <ul>
     <% @lessons.each do |lesson| %>
-       <li>
-          <p class="text-1xl text-center m-3 text-gray-600 dark:text-gray-300">
-            <%= link_to lesson.title, lesson_path(lesson), class: 'underline hover:no-underline hover:text-gray-500 dark:hover:text-gray-400' %>
-          </p>
-        </li>
+      <li class="max-w-md mx-auto my-4 rounded-lg shadow transition bg-white hover:bg-gray-100 dark:bg-gray-800 dark:hover:bg-gray-700">
```
Comment (KevinMulhern) [inline app/views/lessons/installation_lessons/index.html.erb:7]: This is a lot like our card styling. We can reuse that here by using our CardCompent to make sure everything stays consistent 💪 

We're better off changing this from an unordered list and just have items within a containing section element which will handle the max width and spacing between the elements:
```erb
<section class="max-w-xl mx-auto space-y-5">
    <% @lessons.each do |lesson| %>
      <%= render CardComponent.new(classes: "transistion dark:hover:bg-gray-700 hover:bg-gray-100") do |card| %>
        <% card.body(classes: "py-4 px-4") do %>
          <%= link_to lesson_path(lesson) do %>
            <div class="flex items-center gap-3">
              <%= inline_svg "icons/book.svg", alt: 'lesson icon', class: 'w-6 dark:text-gray-500 text-gray-400 shrink-0', title: 'Lesson' %>

                <p class=" text-gray-600 dark:text-gray-300">
                  <%= lesson.title %>
                </p>
            </div>
          <% end %>
        <% end %>
      <% end %>
    <% end %>
  </section>
```

**Review by KevinMulhern — CHANGES_REQUESTED** [review]
Sorry for the delay, nice work with this, it already looks so much better. Just some feedback on how we can reuse some existing styling. Happy to discuss anything.

File: app/views/lessons/installation_lessons/index.html.erb (hunk 1 above)
Comment (wzhengg) [reply app/views/lessons/installation_lessons/index.html.erb:7]: Thanks for the feedback, @KevinMulhern! I'll try to have the changes made by the end of the week.

Comment (wzhengg) [reply app/views/lessons/installation_lessons/index.html.erb]: I've made the proposed changes.

File: app/views/lessons/installation_lessons/index.html.erb [hunk 2]
```diff
@@ -1,14 +1,21 @@
...
+<div class="page-container min-h-screen">
+  <h1 class="page-heading-title text-center">Installation Guides</h1>
+  <section class="max-w-xl mx-auto space-y-5">
     <% @lessons.each do |lesson| %>
-       <li>
-          <p class="text-1xl text-center m-3 text-gray-600 dark:text-gray-300">
-            <%= link_to lesson.title, lesson_path(lesson), class: 'underline hover:no-underline hover:text-gray-500 dark:hover:text-gray-400' %>
-          </p>
-        </li>
+      <%= render CardComponent.new(classes: "transistion dark:hover:bg-gray-700 hover:bg-gray-100") do |card| %>
```
Comment (ChargrilledChook) [inline app/views/lessons/installation_lessons/index.html.erb:7]: Incredibly minor, but there's a linting issue here that is causing the CI tests to fail. Easy fix, just need to swap the double quotes for single quotes.

You can click on these links in the PR to see the details:

image

and you can run `bin/test` from the app root locally to run all the linters and tests that are used in CI

Comment (wzhengg) [reply app/views/lessons/installation_lessons/index.html.erb]: Thanks for letting me know, @ChargrilledChook. It's passing all tests now.

**Review by KevinMulhern — APPROVED** [review]
This looks great @wzhengg, huge improvement 🏅 
Thanks for all your work on this 💪

**Comment by wzhengg** [comment]
Thanks for guiding me through my first contribution, @KevinMulhern!

**Pull request merged**